### ISSUE TITLE
a bunch of our unit tests aren't being run

### ISSUE THREAD

**Issue by binary1230** [magfest/ubersystem#1574]
I was playing around with PyCharm's new code coverage tool and stumbled on some tests that don't appear to be running currently.

In uber/tests/models/test_attendee.py, there are a few functions in the ```TestStaffingAdjustments``` class which do not start with the prefix 'test_' in them, so they are not being run as tests.

I am looking around for others, will submit a PR for re-enabling them.  Sadly it appears that 4 of 6 of these tests are currently failing aaaaaaaaaaaargh.  Will check into this.

Example tests: no_to_yes_ribbon(), no_to_yes_volunteering() should be test_no_to_yes_volunteering(), test_no_to_yes_ribbon()

(sidenote: holy crap that code coverage tool is amazingly useful in the 5 minutes since I've started playing with it. crazytown)

**Comment by binary1230** [comment]
good news is it looks like these 4 tests aren't failing due to logic errors, but just from some kind of bad calling of ```Mock().called```, will check it out.

**Comment by binary1230** [comment]
Here's the failing test:

```python
FAILED

self = <uber.tests.models.test_attendee.TestStaffingAdjustments object at 0x7f50febaf358>
unset_volunteering = <Mock id='139985847779056'>

    def test_no_to_yes_ribbon(self, unset_volunteering):
        with Session() as session:
            a = session.attendee(first_name='Regular', last_name='Attendee')
            a.ribbon = c.VOLUNTEER_RIBBON
            a._staffing_adjustments()
            assert a.staffing
>           assert not unset_volunteering.called

sideboard/plugins/uber/uber/tests/models/test_attendee.py:323: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sideboard/sideboard/lib/sa/__init__.py:199: in __exit__
    self.session.commit()
sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py:790: in commit
    self.transaction.commit()
sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py:392: in commit
    self._prepare_impl()
sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py:372: in _prepare_impl
    self.session.flush()
sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py:2004: in flush
    self._flush(objects)
sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py:2031: in _flush
    self.dispatch.before_flush(self, flush_context, objects)
sideboard/env/lib/python3.4/site-packages/sqlalchemy/event/attr.py:218: in __call__
    fn(*args, **kw)
sideboard/plugins/uber/uber/models.py:1825: in _presave_adjustments
    model.presave_adjustments()
sideboard/plugins/uber/uber/models.py:163: in presave_adjustments
    self._invoke_adjustment_callbacks('presave_adjustment')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Attendee full_name='Regular Attendee'>, label = 'presave_adjustment'

    def _invoke_adjustment_callbacks(self, label):
        callbacks = []
        for name, attr in self._class_attrs.items():
            if hasattr(attr, '__call__') and hasattr(attr, label):
                callbacks.append(getattr(self, name))
>       callbacks.sort(key=lambda f: getattr(f, label))
E       TypeError: unorderable types: Mock() < int()

sideboard/plugins/uber/uber/models.py:158: TypeError
```

**Comment by binary1230** [comment]
update: ```assert not unset_volunteering.called``` is not what's actually failing here, it's the ```with Session()``` is calling session.commit() which is invoking ```_invoke_adjustment_callbacks``` and that's getting confused by our Mock() insertion.

**Comment by binary1230** [comment]
first round of fixes are in https://github.com/magfest/ubersystem/pull/1575/

that MIGHT be it though I haven't had a chance to review the rest of the unit tests.  will look around more thoroughly.

**Comment by binary1230** [comment]
ok, as far as I can tell, it was just these couple of tests, so this is fixed by #1575
